### PR TITLE
Add a name question

### PR DESCRIPTION
**Pull request by felixclack**
We want to collect a person's name to help match their TRN.

Previous names can be helpful in the matching process too but are
optional.

Assumptions:
* no validations required for the previous name fields

### Checklist

- [x] Attach to Trello card
- [x] Rebased master
- [x] Cleaned commit history
- [x] Tested by running locally

### CODE REVIEW

File: app/views/trn_requests/show.html.erb [hunk 1]
```diff
@@ -7,6 +7,11 @@
     <p class="govuk-body">We will send your TRN to <%= @trn_request.email %> if we find one matching your details.</p>
 
     <% rows = [
+        { 
+          key: { text: 'Your Name' },
```
Comment (tvararu) [inline app/views/trn_requests/show.html.erb:11]:
```suggestion
          key: { text: 'Name' }, 
```

File: app/views/trn_requests/show.html.erb [hunk 2]
```diff
@@ -7,6 +7,11 @@
     <p class="govuk-body">We will send your TRN to <%= @trn_request.email %> if we find one matching your details.</p>
 
     <% rows = [
+        { 
+          key: { text: 'Your Name' }, 
+          value: { text: @trn_request.name }, 
+          actions: [{ href: name_path, visually_hidden_text: 'name' }]
+        },
```
Comment (tvararu) [inline app/views/trn_requests/show.html.erb:14]: Could we also play back `Previous name` here as a separate section?

File: app/forms/name_form.rb [hunk 3]
```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+class NameForm
+  include ActiveModel::Model
+
+  attr_accessor :trn_request
+  attr_writer :first_name, :previous_first_name, :previous_last_name, :last_name
+
+  validates :first_name, presence: true
+  validates :last_name, presence: true
```
Comment (tvararu) [inline app/forms/name_form.rb:9]: Could we add a max length validation of 255 (+ copy the message from another form) to all 4 fields here?

File: app/views/name/edit.html.erb [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+      <%= f.govuk_text_field(:first_name,
+            label: { size: 'm', text: 'First name' },
+            required: true
+          ) %>
+      <%= f.govuk_text_field(:last_name,
+            label: { size: 'm', text: 'Last name' },
+            required: true
+          ) %>
+
+      <%= f.govuk_check_boxes_fieldset :name_changed, legend: nil do %>
```
Comment (tvararu) [inline app/views/name/edit.html.erb:21]: Found a small bug locally; when clicking the `Change name` link on the check answers page when I've entered a previous name, the checkbox renders unticked, so the previous names are hidden on initial page load.

**Review by tvararu — COMMENTED** [review]
Looking really good! Ready to approve after the suggested tweaks.

File: app/forms/name_form.rb (hunk 3 above)
Comment (tvararu) [reply app/forms/name_form.rb:9]: @fofr I read your design history post at https://tra-digital-design-history.herokuapp.com/find-a-lost-trn/names/

One suggested example is that the name "Psy", with no first/last name, should actually be a valid "name."

Is that the case in the MVP? Or one to address later if we decide to move to a combined single `Name` field?

Comment (fofr) [reply app/forms/name_form.rb:9]: Kate Lee had some stats on how often this occurred – while there are some records with no first or last names (more commonly first than last), it represents <0.01% in under 65s. So not in MVP.

If we track the validation messages we can see how frequently people attempt to enter only one name.

Comment (felixclack) [reply app/forms/name_form.rb]: 👍

File: app/views/name/edit.html.erb (hunk 4 above)
Comment (felixclack) [reply app/views/name/edit.html.erb:21]: Fixed. 👍

File: app/views/trn_requests/show.html.erb (hunk 2 above)
Comment (felixclack) [reply app/views/trn_requests/show.html.erb:14]: Done.

Screen Shot 2022-03-11 at 8 59 04 am

Comment (tvararu) [reply app/views/trn_requests/show.html.erb:14]: Nice, could they be combined into just `Previous name`?

Comment (felixclack) [reply app/views/trn_requests/show.html.erb:14]: Sure. So, if only one previous name field is present, combine it with the other existing name to get a full name?

eg. Joan Smith enters a previous last name of Doe, so we display Joan Doe.

Comment (fofr) [reply app/views/trn_requests/show.html.erb:14]: I like this approach

**Review by tvararu — APPROVED** [review]

**Pull request merged**